### PR TITLE
test(sdk): Add a new test util for spinning up anvil and test clients

### PR DESCRIPTION
**Pull request by roninjin10**
- Adds a new test client for interacting with anvil in sdk tests
- Need this in follow up pr to TDD add support for proving multiple withdrawals
- It's used in gateway atm but if we use it in more packages we should seperate this into it's own reusable package

To start anvil

```typescript
await anvil.start()
```

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6006** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6006" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6006?utm_source=stack-comment).

**Comment by roninjin10** [comment]
@jakim929 we are chilling we already got forge installed as it's used to build contracts-bedrock

**Comment by roninjin10** [comment]
no idea if CI is broken or my yarn install broke CI

**Comment by roninjin10** [comment]
CI is actually broken.  Our npm packages are in shambles

**Comment by roninjin10** [comment]
Blocked by npm packages atm

**Comment by fubhy** [comment]
I think what you want here is:

1. A global setup file for vite.config.ts (globalSetup).
2. In that, spin up two(!) anvil proxies with the two different configurations you need for your test suite
3. Connect to those proxies from your test files using the `VITE_WORKER_ID` for the anvil instance.

For 3. it makes sense to create a cute util function that gives you the Viem client instances for the current context.

So basically just like https://github.com/wagmi-dev/anvil.js/tree/main/examples/example-vitest but with two proxies because you need two different underlying anvil instances if I'm interpreting your requirements correctly.

@roninjin10 Happy to help out here but need a bit more context / understanding what your goal is.

**Comment by roninjin10** [comment]
Great advice!

**Comment by roninjin10** [comment]
This was blocked by a node module installation issue in past. We just worked around it though.  Now that we use pnpm without hoisting node modules that issue should be gone. Will land this pr next time we add more tests
